src/GeneratedMapper/Information/MappingInformation.cs

```diff
@@ -13,17 +13,59 @@ internal sealed class MappingInformation
         private readonly List<Diagnostic> _diagnostics = new();
         private readonly List<PropertyMappingInformation> _propertyMappings = new();
 
-        public AttributeData AttributeData { get; private set; }
+        public const int MapToIndex = int.MinValue;
+
+        public SyntaxReference SyntaxReference { get; }
+        public int? MaxRecursion { get; }
         public ConfigurationValues ConfigurationValues { get; private set; }
         public int AttributeIndex { get; private set; }
 
-        public MappingInformation(AttributeData attributeData, ConfigurationValues configurationValues)
+        public MappingInformation(SyntaxReference syntaxReference, int? maxRecursion, int index, ConfigurationValues configurationValues)
         {
-            AttributeData = attributeData;
+            SyntaxReference = syntaxReference;
+            MaxRecursion = maxRecursion;
             ConfigurationValues = configurationValues;
-            AttributeIndex = attributeData.GetIndex();
+            AttributeIndex = index;
+        }
+
+        private sealed class SourceTypeDestinationTypeEqualityComparer : IEqualityComparer<MappingInformation>
+        {
+            public bool Equals(MappingInformation x, MappingInformation y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                if (ReferenceEquals(x, null))
```

Comment:
This looks a bit silly. Isn't a `if (x == null || y == null) { return false; }` more easy?

Reply:
Resharper made this code and got the job done.  I'll worry about looks and placement in the refactor.

Reply:
Fixed